.github/workflows/deploy-stats-api.yml

```diff
@@ -47,56 +45,81 @@ jobs:
       - name: Format check
         run: terraform fmt -check
 
-  build-push:
+  deploy:
     runs-on: ubuntu-latest
-    environment: ${{ inputs.environment || 'development' }}
-
+    needs: format-check
+    environment: ${{ inputs.env }}
+    if: >
+      (github.ref == 'refs/heads/main' || github.event_name == 'workflow_dispatch') &&
+      ((inputs.env != 'production' && inputs.env != 'prod' && inputs.env != 'staging' && inputs.env != 'stage') || inputs.image_tag != 'latest')
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Authenticate to Google Cloud
+      - name: Set environment variables
+        run: |
+          # Determine environment from inputs or GitHub environment
+          if [ -n "${{ inputs.env }}" ]; then
+            ENV_NAME="${{ inputs.env }}"
+          else
+            echo "Error: No environment specified"
+            exit 1
+          fi
+          
+          REGION="${{ vars.GCP_REGION || 'us-central1' }}"
+          echo "ENVIRONMENT=$ENV_NAME" >> $GITHUB_ENV
+          echo "REGION=$REGION" >> $GITHUB_ENV
+
+          # Set project name and terraform bucket based on environment
+          if [ "$ENV_NAME" = "production" ] || [ "$ENV_NAME" = "prod" ]; then
```

Comment:
Very confusing? Above we have `inputs.env != 'production' && inputs.env != 'prod' ...` as condition?